File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.js
/**
 * @param {string} s
 * @param {number} k
 * @param {character} fill
 * @return {string[]}
 */
var divideString = function (s, k, fill) {
    const ans = []
    const n = s.length

    let curr = 0

    while (curr < n) {
        const end = Math.min(curr + k, n)
        ans.push(s.slice(curr, end))
        curr += k
    }
    const lastString = ans[ans.length - 1]
    if (lastString.length < k) {
        ans[ans.length - 1] = lastString + fill.repeat(k - lastString.length)
    }
    return ans
};

File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.java
class Solution {

    public String[] divideString(String s, int k, char fill) {
        List<String> ans = new ArrayList<>(); // grouped string
        
        int n = s.length();
        int curr = 0; // starting index of each group
        // split string
        while (curr < n) {
            int end = Math.min(curr + k, n);
            ans.add(s.substring(curr, end));
            curr += k;
        }
        // try to fill in the last group
        String last = ans.get(ans.size() - 1);
        if (last.length() < k) {
            last += String.valueOf(fill).repeat(k - last.length());
            ans.set(ans.size() - 1, last);
        }
        return ans.toArray(new String[0]);
    }
}